libminifi/test/aws-tests/FetchS3ObjectTests.cpp

```diff
@@ -0,0 +1,174 @@
+/**
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <fstream>
+
+#include "S3TestsFixture.h"
+#include "processors/FetchS3Object.h"
+
+class FetchS3ObjectTestsFixture : public S3TestsFixture<minifi::aws::processors::FetchS3Object> {
+ public:
+  FetchS3ObjectTestsFixture() {
+    auto putfile = plan->addProcessor(
+      "PutFile",
+      "PutFile",
+      core::Relationship("success", "d"),
+      true);
+    char output_dir_mask[] = "/tmp/gt.XXXXXX";
+    output_dir = test_controller.createTempDirectory(output_dir_mask);
```

Comment:
You can get rid of the mask by using:
```c++
output_dir = utils::createTempDir(&testController);
```

Reply:
Added in [7b57ab6](https://github.com/apache/nifi-minifi-cpp/pull/970/commits/7b57ab6c5a077ae14d490eb325a3f049bf629d68)